### PR TITLE
Fix leaking thread from ClientConsoleTest

### LINKED ISSUE #17296: ClientConsoleApp in ClientConsoleTest is not shutdown sometimes
Test `com.hazelcast.client.console.ClientConsoleTest.connectsToHazelcastCluster()` uses `ClientConsoleApp` which is sometimes not shutdown and stay alive even after test is finished. It blocks finishing surefire tests (it is terminated by surefire after 30 seconds when all tests are finished) which for example causes that jacoco plugin is not executed correctly and test coverage is not reported.

Related thread dump:
```
"pool-7123-thread-1" #160189 prio=5 os_prio=0 tid=0x00007f177c16a000 nid=0x1603 runnable [0x00007f17638f9000]
   java.lang.Thread.State: RUNNABLE
	at java.io.FileInputStream.readBytes(Native Method)
	at java.io.FileInputStream.read(FileInputStream.java:255)
	at java.io.BufferedInputStream.read1(BufferedInputStream.java:284)
	at java.io.BufferedInputStream.read(BufferedInputStream.java:345)
	- locked <0x000000008006e260> (a java.io.BufferedInputStream)
	at sun.nio.cs.StreamDecoder.readBytes(StreamDecoder.java:284)
	at sun.nio.cs.StreamDecoder.implRead(StreamDecoder.java:326)
	at sun.nio.cs.StreamDecoder.read(StreamDecoder.java:178)
	- locked <0x0000000082ae1530> (a java.io.InputStreamReader)
	at java.io.InputStreamReader.read(InputStreamReader.java:184)
	at java.io.BufferedReader.fill(BufferedReader.java:161)
	at java.io.BufferedReader.readLine(BufferedReader.java:324)
	- locked <0x0000000082ae1530> (a java.io.InputStreamReader)
	at java.io.BufferedReader.readLine(BufferedReader.java:389)
	at com.hazelcast.client.console.ClientConsoleApp$DefaultLineReader.readLine(ClientConsoleApp.java:185)
	at com.hazelcast.client.console.ClientConsoleApp.start(ClientConsoleApp.java:169)
	at com.hazelcast.client.console.ClientConsoleApp.main(ClientConsoleApp.java:1551)
	at com.hazelcast.client.console.ClientConsoleTest.lambda$connectsToHazelcastCluster$0(ClientConsoleTest.java:91)
	at com.hazelcast.client.console.ClientConsoleTest$$Lambda$3794/200997726.run(Unknown Source)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
```

It can be reproduced by adding `Thread.sleep(60_000);` after `assertNotNull(client);` in given test. You can also add `<forkedProcessExitTimeoutInSeconds>900</forkedProcessExitTimeoutInSeconds>` to `configuration` of surefire plugin to wait 15 minutes instead of default 30 seconds after all tests are finished. If you dump threads during this period (after tests are finished) you'll see mentioned above thread.

### PR DESCRIPTION
**Pull request by kwart**
Fixes #17296.

### CODE REVIEW

**Comment by hz-highfive** [comment]
The job `Hazelcast-pr-builder` of your PR failed ([log](
https://j-artifacts.s3.amazonaws.com/Hazelcast-pr-builder/1680/build_log.txt), [artifacts](https://j-artifacts.s3.amazonaws.com/Hazelcast-pr-builder/1680/Hazelcast-pr-builder-1680.zip)). 
Through arcane magic we have determined that the following fragments from the build log may contain information about the problem.
<details>
<summary>Click to expand the log file</summary>
<pre>
--------------------------
---------SUMMARY----------
--------------------------
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-checkstyle-plugin:3.1.1:checkstyle (default) on project hazelcast: An error has occurred in Checkstyle report generation. Failed during checkstyle execution: There is 1 error reported by Checkstyle 8.29 with /home/jenkins/jenkins_slave/workspace/Hazelcast-pr-builder/checkstyle/checkstyle.xml ruleset. -> [Help 1]
--------------------------
---------ERRORS-----------
--------------------------
[ERROR] /home/jenkins/jenkins_slave/workspace/Hazelcast-pr-builder/hazelcast/src/test/java/com/hazelcast/client/console/ClientConsoleTest.java:49:8: Unused import - com.hazelcast.internal.util.StringUtil. [UnusedImports]
--------------------------
</details>
</pre>

**Comment by hz-highfive** [comment]
The job `Hazelcast-pr-builder` of your PR failed ([log](
https://j-artifacts.s3.amazonaws.com/Hazelcast-pr-builder/1682/build_log.txt), [artifacts](https://j-artifacts.s3.amazonaws.com/Hazelcast-pr-builder/1682/Hazelcast-pr-builder-1682.zip)). 
Through arcane magic we have determined that the following fragments from the build log may contain information about the problem.
<details>
<summary>Click to expand the log file</summary>
<pre>
--------------------------
---------SUMMARY----------
--------------------------
[ERROR] Failed to execute goal com.github.spotbugs:spotbugs-maven-plugin:3.1.12:check (default) on project hazelcast: failed with 1 bugs and 0 errors -> [Help 1]
--------------------------
</details>
</pre>

File: hazelcast/src/test/java/com/hazelcast/client/console/ClientConsoleTest.java [hunk 1]
```diff
@@ -71,7 +76,7 @@ public void after() {
      * YAML client configuration file.
      */
     @Test
-    public void connectsToHazelcastCluster() throws IOException {
+    public void connectsToHazelcastCluster() throws IOException, InterruptedException {
```
Comment (olukas) [inline hazelcast/src/test/java/com/hazelcast/client/console/ClientConsoleTest.java:79]: Throwing `InterruptedException` is not needed here.

Comment (kwart) [reply hazelcast/src/test/java/com/hazelcast/client/console/ClientConsoleTest.java:79]: :+1: Nice spot. I'll fix it. It was introduced by playing with a `Thread.sleep` within the test.

**Comment by kwart** [comment]
Thanks for the reviews, guys. Merging.

**Pull request merged**